imagedephi/models/rules.py

```diff
@@ -11,27 +11,38 @@ class FileFormat(Enum):
     SVS = "svs"
 
 
-class _MetadataRule(BaseModel):
+class _Rule(BaseModel):
     # key_name is not set by users, but is availible internally
     key_name: str = Field(exclude=True)
     action: Literal["keep", "delete", "replace"]
 
 
-class ReplaceMetadataRule(_MetadataRule):
+class KeepRule(_Rule):
+    action: Literal["keep"]
+
+
+class DeleteRule(_Rule):
+    action: Literal["delete"]
+
+
+class ReplaceRule(_Rule):
     action: Literal["replace"]
-    new_value: str
 
 
-class KeepMetadataRule(_MetadataRule):
-    action: Literal["keep"]
+class ReplaceMetadataRule(ReplaceRule):
+    new_value: str
 
 
-class DeleteMetadataRule(_MetadataRule):
-    action: Literal["delete"]
+class ReplaceImageRule(_Rule):
+    replace_with: Literal["black_square"]
```

Comment:
Found an issue. This should inherit from `ReplaceRule`.

I imagine that `keep` and `delete` rules will work the same for all images and metadata, as there's no room for interpretation. However, `ReplaceRule`s will be different depending on what they are applied to (replace metadata with string, replace an image with a different image, etc.)